src/rules.jl

```diff
@@ -93,6 +95,133 @@ julia> hypot_pullback(1) == (NO_FIELDS, (x / hypot(x, y)), (y / hypot(x, y)))
 true
 ```
 
+When defining overloads they should be wrapped with the [`@rrule`](@ref) macro.
+
 See also: [`frule`](@ref), [`@scalar_rule`](@ref)
 """
 rrule(::Any, ::Vararg{Any}; kwargs...) = nothing
+
+"""
+    @frule(function ...)
+
+[`frule`](@ref) defining functions should be decorated with this macro.
+
+Example:
+```julia
+@frule function frule((Δself, Δargs...), ::typeof(foo), args...; kwargs...)
+    ...
+    return y, ∂y
+end
+```
+"""
+macro frule(ast)
+    return quote
+        $(esc(ast))
+        $_register_new_rule(frule, $(QuoteNode(ast)))
+    end
+end
+
+"""
+    @rrule(function ...)
+
+[`rrule`](@ref) defining functions should be decorated with this macro.
```

Comment:
```suggestion
New methods of [`rrule`](@ref) should be decorated with this macro.
```

Reply:
I am not going to apply this change til we decide if we are even keeping the decorator macros.
If we are not then it is simpler to not add more commits to revert